pkg/monitor/monitorapi/construction.go

```diff
@@ -347,7 +360,18 @@ func (b *LocatorBuilder) WithAPIUnreachableFromClient(metric model.Metric, servi
 	}
 
 	b.targetType = LocatorTypeAPIUnreachableFromClient
-	b.annotations[LocatorAPIUnreachableHostKey] = getHost(metric, serviceNetworkIP)
+	if host := getHost(metric, resolver.GetKubernetesServiceClusterIP()); len(host) > 0 {
+		b.annotations[LocatorAPIUnreachableHostKey] = host
+	}
+	if job := string(metric["job"]); len(job) > 0 {
+		b.annotations[LocatorAPIUnreachableComponentKey] = job
+	}
+
+	// we try to get the node name from the instance label
+	if nodeName, nodeRole, err := resolver.GetNodeNameAndRoleFromInstance(string(metric["instance"])); err == nil {
+		b.annotations[LocatorNodeKey] = nodeName
```

Comment:
For reuse-ability, consider extracting these out as a LocatorBuilder methods. e.g.:
```
func (b *LocatorBuilder) NodeFromIP(ip string) Locator
func (b *LocatorBuilder) NodeFromIPWithNodeRole(ip string) Locator
```